pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -597,13 +599,65 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 	return err
 }
 
+func isSidecar(containers []v1.Container, name string) bool {
+	for _, container := range containers {
+		if container.Name == name && lifecycleSidecar(container) {
+			return true
+		}
+	}
+	return false
+}
+
+func gracePeriodRemaining(pod *v1.Pod, gracePeriodOverride *int64, startTime time.Time) *int64 {
+	timeElapsed := int64(time.Since(startTime).Seconds())
+
+	switch {
+	case gracePeriodOverride != nil:
+		newPeriod := *gracePeriodOverride - timeElapsed
+		gracePeriodOverride = &newPeriod
+	case pod.DeletionGracePeriodSeconds != nil:
+		newPeriod := *pod.DeletionGracePeriodSeconds - timeElapsed
+		gracePeriodOverride = &newPeriod
+	case pod.Spec.TerminationGracePeriodSeconds != nil:
+		newPeriod := *pod.Spec.TerminationGracePeriodSeconds - timeElapsed
+		gracePeriodOverride = &newPeriod
+	}
+
+	return gracePeriodOverride
+}
+
 // killContainersWithSyncResult kills all pod's containers with sync results.
 func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, runningPod kubecontainer.Pod, gracePeriodOverride *int64) (syncResults []*kubecontainer.SyncResult) {
 	containerResults := make(chan *kubecontainer.SyncResult, len(runningPod.Containers))
 	wg := sync.WaitGroup{}
 
-	wg.Add(len(runningPod.Containers))
-	for _, container := range runningPod.Containers {
+	var containers []*kubecontainer.Container
```

Comment:
```suggestion
	containers := runningPod.Containers
```
This removes the need for the else checks below.

Reply:
fixed